### PR TITLE
Creating app with input output

### PR DESCRIPTION
**Pull request by AlexMelov**

### CODE REVIEW

File: frontend/src/app/app.component.ts [hunk 1]
```diff
@@ -4,17 +4,18 @@ import { Todo } from './todo.intercace';
...
 @Component(
-{
-	selector: 'app-root',
-	templateUrl: './app.component.html',
-	styleUrls:
-	[
-		'./app.component.css'
-	]
-})
+	{
```
Comment (henryruhs) [inline frontend/src/app/app.component.ts]: A kind of wrong indent...

File: frontend/src/app/list/list.component.css [hunk 2]
```diff
@@ -1,11 +1,38 @@
...
+	justify-content: space-between;
 	list-style-type: none;
-	margin: 0.5rem 0;
-	padding: 0.25rem;
+	margin: 1rem 0;
+	min-width: 50%;
+	padding: 0.25rem 1rem 0.25rem 1rem;
+	transition: all 0.3s ease;
+}
+.list-item button
```
Comment (henryruhs) [inline frontend/src/app/list/list.component.css]: Missing newline

File: frontend/src/app/list/list.component.css [hunk 3]
```diff
@@ -1,11 +1,38 @@
...
+	background-color: #3057f1;
+	border: none;
+	border-radius: 0.3rem;
+	color: #ffffff;
+	height: 3vh;
+	outline: none;
+	transition: all 0.3s ease;
+	width: 8rem;
+}
+.list-item button:hover
```
Comment (henryruhs) [inline frontend/src/app/list/list.component.css]: Missinh newline

File: frontend/src/app/list/list.component.css (hunk 2 above)
Comment (henryruhs) [inline frontend/src/app/list/list.component.css]: Why don't you use nesting in the whole file?

File: frontend/src/app/list/list.component.css [hunk 4]
```diff
@@ -1,11 +1,38 @@
 .list-item
```
Comment (henryruhs) [inline frontend/src/app/list/list.component.css]: Switch to SCSS

File: frontend/src/app/list/list.component.ts [hunk 5]
```diff
@@ -1,24 +1,20 @@
-import { Component, Input } from '@angular/core';
-import { environment } from '../../environments/environment';
-import { HttpClient } from '@angular/common/http';
+import { Component, Input, Output, EventEmitter } from '@angular/core';
 import { Todo } from '../todo.intercace';
 
 @Component({
```
Comment (henryruhs) [inline frontend/src/app/list/list.component.ts]: { on newline...

File: frontend/src/app/new-todo/new-todo.component.css [hunk 6]
```diff
@@ -0,0 +1,38 @@
+.form-text-input input
+{
+	border: none;
```
Comment (henryruhs) [inline frontend/src/app/new-todo/new-todo.component.css]: You can use a css reset of mine: https://github.com/henryruhs/harmonize.css

After that there is no need to reset elements with border: 0 and so on

File: frontend/src/app/new-todo/new-todo.component.ts [hunk 7]
```diff
@@ -1,25 +1,23 @@
-import { Component } from '@angular/core';
-import { HttpClient } from '@angular/common/http';
+import { Component, Output, EventEmitter } from '@angular/core';
 import { Todo } from '../todo.intercace';
-import { environment } from '../../environments/environment';
 
 @Component({
```
Comment (henryruhs) [inline frontend/src/app/new-todo/new-todo.component.ts]: { on newline

File: frontend/src/app/new-todo/new-todo.component.ts [hunk 8]
```diff
@@ -1,25 +1,23 @@
...
 
-	constructor(private http : HttpClient)
-	{
-	}
-
 	addNewTodo() : void
 	{
-		this.http.post(environment.apiUrl + environment.apiRoutes.todos, this.todo)
-			.subscribe();
+		this.todoCreated.emit(
```
Comment (henryruhs) [inline frontend/src/app/new-todo/new-todo.component.ts:17]: Like that:

```
this.todoCreated.emit(
{
	title: this.todo.title
});
```

**Review by henryruhs — CHANGES_REQUESTED** [review]

File: frontend/src/app/app.module.ts [hunk 9]
```diff
@@ -9,17 +9,17 @@ import { NewTodoComponent } from './new-todo/new-todo.component';
 
 @NgModule({
 	declarations:
-	[
-		AppComponent,
-		ListComponent,
-		NewTodoComponent
-	],
+		[
```
Comment (henryruhs) [inline frontend/src/app/app.module.ts]: Wrong indent

File: frontend/src/app/list/list.component.spec.ts [hunk 10]
```diff
@@ -1,7 +1,11 @@
-describe('running test for list component', () =>
+describe('Async component', ()=>
 {
-	it('should run empty test', () =>
+	test('render list items with mocks', async() =>
 	{
+		window.fetch = jest.fn().mockResolvedValueOnce({
```
Comment (henryruhs) [inline frontend/src/app/list/list.component.spec.ts]: we dont use fetch in Angular

File: frontend/src/app/list/list.component.ts [hunk 11]
```diff
@@ -1,24 +1,22 @@
...
-  selector: 'app-list',
-  templateUrl: './list.component.html',
-  styleUrls: [ './list.component.css' ]
-})
+@Component(
+	{
+		selector: 'app-list',
+		templateUrl: './list.component.html',
+		styleUrls:
+			[
```
Comment (henryruhs) [inline frontend/src/app/list/list.component.ts]: Wrong indent

File: frontend/src/app/new-todo/new-todo.component.scss [hunk 12]
```diff
@@ -0,0 +1,34 @@
...
+		color: #000000;
+		cursor: pointer;
+		font-weight: 500;
+		height: 3vh;
+		margin-left: 2rem;
+		transition: all 0.3s ease;
+		width: 10rem;
+	}
+
+	button:hover
```
Comment (henryruhs) [inline frontend/src/app/new-todo/new-todo.component.scss]: Could be nested &:hover

File: frontend/src/app/new-todo/new-todo.component.ts [hunk 13]
```diff
@@ -1,25 +1,26 @@
...
-  	selector: 'app-new-todo',
-  	templateUrl: './new-todo.component.html',
-  	styleUrls: [ './new-todo.component.css' ]
-})
+@Component(
+	{
+		selector: 'app-new-todo',
+		templateUrl: './new-todo.component.html',
+		styleUrls:
+			[
```
Comment (henryruhs) [inline frontend/src/app/new-todo/new-todo.component.ts]: Wrong indent

File: frontend/src/app/todo.intercace.ts [hunk 14]
```diff
@@ -2,4 +2,5 @@ export interface Todo
 {
 	title ?: string | null;
 	_id ?: string | null;
+	__v ?: number | null;
```
Comment (henryruhs) [inline frontend/src/app/todo.intercace.ts]: What is __v? such a naming looks horrible to me

File: frontend/src/environments/environment.local.ts [hunk 15]
```diff
@@ -0,0 +1,18 @@
+import { Environment } from './environment.interface';
+
+export const environment : Environment =
+	{
+		metadata:
+			{
```
Comment (henryruhs) [inline frontend/src/environments/environment.local.ts]: Wrong indent

File: frontend/src/environments/environment.local.ts [hunk 16]
```diff
@@ -0,0 +1,18 @@
...
+				branch: 'develop',
+				environment: 'local'
+			},
+		baseUrl : 'http://localhost:4200',
+		apiUrl: 'http://localhost:8000',
+		apiRoutes:
+			{
```
Comment (henryruhs) [inline frontend/src/environments/environment.local.ts]: Wrong indent

File: frontend/src/styles.css [hunk 17]
```diff
@@ -0,0 +1 @@
+@import "~harmonize.css/harmonize.css";
```
Comment (henryruhs) [inline frontend/src/styles.css]: Single quotes

File: .github/workflows/ci-dev.yml [hunk 18]
```diff
@@ -4,6 +4,7 @@ on:
   push:
     branches:
       - develop
+      - creating-app-with-input-output
```
Comment (henryruhs) [inline .github/workflows/ci-dev.yml]: Remove the branch from pipeline

File: frontend/src/app/app.component.scss [hunk 19]
```diff
@@ -0,0 +1,3 @@
+*{
```
Comment (henryruhs) [inline frontend/src/app/app.component.scss]: The universal selector is a bad thing, avoid using it... you either want to set the body tag but properly not in this file

File: frontend/src/app/app.component.ts [hunk 20]
```diff
@@ -1,20 +1,22 @@
...
+		selector: 'app-root',
+		templateUrl: './app.component.html',
+		styleUrls:
+			[
```
Comment (henryruhs) [inline frontend/src/app/app.component.ts]: Wrong indent

File: frontend/src/app/app.module.ts [hunk 21]
```diff
@@ -9,17 +9,17 @@ import { NewTodoComponent } from './new-todo/new-todo.component';
...
+			ListComponent,
+			NewTodoComponent
+		],
 	imports:
-	[
-		BrowserModule,
-		FormsModule,
-		HttpClientModule
-	],
+		[
```
Comment (henryruhs) [inline frontend/src/app/app.module.ts]: Wrong indent

File: frontend/src/app/list/list.component.scss [hunk 22]
```diff
@@ -0,0 +1,40 @@
+.list-item
+{
+	align-items: center;
+	border-radius: 0.3rem;
+	box-shadow: 0 0 0.5rem 0.1rem #b7b7d2;
+	color: #000000;
+	display: flex;
+	font-family: sans-serif;
```
Comment (henryruhs) [inline frontend/src/app/list/list.component.scss]: Why do you set the same font family here again? It has been set global

File: frontend/src/app/list/list.component.scss [hunk 23]
```diff
@@ -0,0 +1,40 @@
...
+	font-size: 1.125rem;
+	font-weight: 600;
+	justify-content: space-between;
+	list-style-type: none;
```
Comment (henryruhs) [inline frontend/src/app/list/list.component.scss]: list style not is not needed when using harmonize.css

File: frontend/src/app/list/list.component.scss [hunk 24]
```diff
@@ -0,0 +1,40 @@
...
+	margin: 1rem 0;
+	min-width: 50%;
+	padding: 1rem;
+	transition: all 0.3s ease;
```
Comment (henryruhs) [inline frontend/src/app/list/list.component.scss]: try to avoid all... use specific like color

File: frontend/src/app/list/list.component.scss [hunk 25]
```diff
@@ -0,0 +1,40 @@
...
+		background-color: #3057f1;
+		border-radius: 0.3rem;
+		color: #ffffff;
+		font-weight: 400;
+		height: 3vh;
+		transition: all 0.3s ease;
+		width: 8rem;
+	}
+
+	button:hover
```
Comment (henryruhs) [inline frontend/src/app/list/list.component.scss]: Use nesting here &:hover

File: frontend/src/app/new-todo/new-todo.component.scss [hunk 26]
```diff
@@ -0,0 +1,34 @@
+.form-text-input input
+{
+	border-bottom: 0.1rem solid #000000;
```
Comment (henryruhs) [inline frontend/src/app/new-todo/new-todo.component.scss]: 0.125rem

File: frontend/angular.json [hunk 27]
```diff
@@ -43,6 +43,17 @@
 					},
 					"configurations":
 					{
+						"local":
```
Comment (henryruhs) [inline frontend/angular.json:46]: Local ist actual not needed... you can properly unset that dev is default

**Review by henryruhs — CHANGES_REQUESTED** [review]

File: frontend/src/app/app.component.ts [hunk 28]
```diff
@@ -1,20 +1,22 @@
...
+				'./app.component.scss'
+			],
+		encapsulation:ViewEncapsulation.None
```
Comment (henryruhs) [inline frontend/src/app/app.component.ts]: Remove encapsulation:ViewEncapsulation.None

File: frontend/src/app/todo.intercace.ts (hunk 14 above)
Comment (AlexMelov) [reply frontend/src/app/todo.intercace.ts]: it is autogenerated from mongoDb and always have value=0  and if I not set this,  it throws error on the emit

**Review by henryruhs — APPROVED** [review]

**Pull request merged**